pytest-embedded-serial-esp/pytest_embedded_serial_esp/serial.py

```diff
@@ -54,6 +67,13 @@ def __init__(
             # sort to make /dev/ttyS* ports before /dev/ttyUSB* ports
             # esptool will reverse the list
             ports.sort()
+            if port_mac:
+                for port in available_ports:
```

Comment:
here's a potential bug. in earlier lines,

```python
            available_ports = esptool.get_port_list()
            ports = list(set(available_ports) - set(self.occupied_ports.keys()))
```

should use `ports` instead. the occupied ports should not be considered.

This potential bug is also happening in PR #141, please also fix it there. sorry I missed that in earlier review.